### PR TITLE
Week 9: Music project Peggy & Max

### PR DESCRIPTION
**Pull request by monkeybusiness101**
Link to my deployed project:

https://goofy-haibt-943dc2.netlify.app/

My reflections on how this week's project turned out:

I had a good collaboration with Peggy and it was a fun project.

Things I'd like to have clarified or explained in more detail:

We have never had any lectures on regex and how to fix the , and & task. We could either get the artists with its URL to work or the , and & but not providing unique links

### CODE REVIEW

File: code/src/components/Artists.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+//Component to show every artist and its URL
+export const Artists = (props) => {
+   return (
+    <span><a className='artists' href={props.artistURL}>
```
Comment (EmOlsz) [inline code/src/components/Artists.js:6]: Indentation could be improved over here

File: code/src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div className='section-container'>
+      <Header />
+      <section className='rows'>
+
+      {/*Mapping through all the albums in the json */}
+      {data.albums.items.map((album, i) => {
+        return (
+            <div className='album-container' key={i}>
+            <Albums key={album.id} url={album.images[0].url} name={album.name} />
```
Comment (EmOlsz) [inline code/src/components/App.js:19]: `key` attribute is considered as required only in root element inside loop. We don't have to add it anywhere else. This snippet of code is wrong :

```jsx
<div key={someUniqueValue}>
  <h1 key={someUniqueValue}>Heading</h1>
  <p key={someUniqueValue}>Paragraph</p>
  <span key={someUniqueValue}>Span</span>
</div>
```

This snippet of code is correct :

```jsx
 <div key={someUniqueValue}>
  <h1>Heading</h1>
  <p>Paragraph</p>
  <span>Span</span>
</div>
```

File: code/src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <ReleaseTitle key={album.name} albumName={album.name} albumURL={album.external_urls.spotify} />
+            <div className='artist-container'>
+              
+                  {/*Mapping through all the artists in the json to get hold of artist name and their respective artist url */}
+                  {album.artists.map((sub, subindex) =>
```
Comment (EmOlsz) [inline code/src/components/App.js:24]: Same here

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Good job Peggy & Max! Your project is well structured and has fine logic. Responsiveness is also done good :)
However, you made some mistakes. Please, take a look on my comments and provide changes, after that I'll approve whole project!

File: code/src/components/Album.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+
+//Component for each album card
+export const Albums = (props) => {
+  return ( 
+    <div className="cover-container">
+        <img src={props.url} alt={props.name} className='album-art'></img>
+        <div className="opacity-layer"></div>
+            <div className="icon-container">
+                <img src="/icons/heart.svg" alt="heart icon" className="heart-icon"></img>
```
[the remark this answers is not in the document]
Comment (monkeybusiness101) [reply code/src/components/Album.js:10]: fixed

File: code/src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import data from '../data.json'
```
[the remark this answers is not in the document]
Comment (monkeybusiness101) [reply code/src/components/App.js:2]: Hi Maks,

Would the data from the json be grouped with local dependencies in this case? Would it be the same scenario if we fetched the data ourselves from API and then saved it in a json file?

I have changed in app.js to:

import React from 'react'

import data from '../data.json'
import { Header } from './Header.js'
import { Albums } from './Album.js'
import { Artists } from './Artists.js'
import { ReleaseTitle } from './ReleaseTitle.js'

and in index.js to:

import React from 'react'
import ReactDOM from 'react-dom'

import { App } from './components/App'

import './index.css'

File: code/src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+export const App = () => {
+  return (
+    <div className='section-container'>
+      <Header />
+      <section className='rows'>
+
+      {/*Mapping through all the albums in the json */}
+      {data.albums.items.map((album, i) => {
+        return (
+            <div className='album-container' key={i}>
```
[the remark this answers is not in the document]
Comment (monkeybusiness101) [reply code/src/components/App.js:18]: I only need to have a key on the div container, correct? Fixed.

File: code/src/components/Artists.js (hunk 1 above)
Comment (monkeybusiness101) [reply code/src/components/Artists.js:6]: fixed

File: code/src/components/App.js (hunk 2 above)
Comment (monkeybusiness101) [reply code/src/components/App.js:19]: fixed and thank you for the valuable feedback

File: code/src/components/App.js (hunk 3 above)
Comment (monkeybusiness101) [reply code/src/components/App.js:24]: fixed

**Comment by monkeybusiness101** [comment]
Hi @EmOlsz! I have made suggested changes to our project. I have one question about the structure of the imported dependencies, hopefully, I understood your comment correct.

File: code/src/components/App.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/components/App.js:3]: Hey Max, exactly! JSON data should be in "local dependencies" group

File: code/src/components/App.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/components/App.js]: Exactly, key is only needed in highest JSX element returned from loop. In this case, it's `<div className='album-container' key={i}>` from line 18

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**